test/parallel_api/ranges/remove_ranges_sycl.pass.cpp

```diff
@@ -0,0 +1,79 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/execution>
+
+#include "support/pstl_test_config.h"
+
+#if _ENABLE_RANGES_TESTING
+#include <oneapi/dpl/ranges>
+#endif
+
+#include "support/utils.h"
+
+#include <vector>
+#include <iostream>
+#include <iterator>
+
+int32_t
+main()
+{
+#if _ENABLE_RANGES_TESTING
+    using T = int;
+    ::std::vector<T> data = {2, 5, 2, 4, 4, 0, 6, -7, 7, 3};
+    const int max_n = data.size();
+
+    auto lambda = [](T val){ return val % 2 == 0; };
+    T val = 2;
+
+    ::std::vector<T> in1(data);
+    ::std::vector<T> in2(data);
+
+    using namespace TestUtils;
+    using namespace oneapi::dpl::experimental::ranges;
+
+    ::std::vector<T>::difference_type in1_end_n;
+    ::std::vector<T>::difference_type in2_end_n;
+    {
+        sycl::buffer<T> A(in1.data(), sycl::range<1>(max_n));
+        sycl::buffer<T> B(in2.data(), sycl::range<1>(max_n));
+
+        using Policy = decltype(TestUtils::default_dpcpp_policy);
+        auto exec = TestUtils::default_dpcpp_policy;
+        auto exec1 = make_new_policy<TestUtils::new_kernel_name<Policy, 0>>(exec);
+        auto exec2 = make_new_policy<TestUtils::new_kernel_name<Policy, 1>>(exec);
+
+        in1_end_n = remove(exec1, all_view<T, sycl::access::mode::read_write>(A), val);
```

Comment:
1) you can use just views::all(A)
2) passing a sycl buffer directly should be tested as well